### PR TITLE
Fix for building with Asan/Ubsan

### LINKED ISSUE #6860: Build fix for Asan/Ubsan
Make it compile again

### PR DESCRIPTION
**Pull request by JCash**
This change enables us to use the Address Sanitizer and Undefined Behavior sanitizer again.

Fixes #6860

### CODE REVIEW

File: .github/workflows/engine-nightly.yml [hunk 1]
```diff
@@ -22,8 +22,12 @@ jobs:
       { name: 'Install Java', uses: actions/setup-java@v3, with: { java-version: '11.0.15', distribution: 'microsoft'} },
       { name: 'Install dependencies', run: 'ci/ci.sh install' },
       {
-        name: 'Build engine',
-        run: 'ci/ci.sh --platform=x86_64-darwin --with-asan --with-vanilla-lua --skip-builtins --skip-docs engine'
+        name: 'ASAN',
+        run: 'ci/ci.sh --platform=x86_64-darwin --with-asan --skip-builtins --skip-docs engine'
```
Comment (JCash) [inline .github/workflows/engine-nightly.yml:26]: Reenabled running with LuaJIT (they seem to work together now).

File: .github/workflows/engine-nightly.yml [hunk 2]
```diff
@@ -41,7 +45,15 @@ jobs:
...
       { name: 'Install dependencies', run: 'ci/ci.sh install' },
       {
-        name: 'Build engine',
+        name: 'ASAN',
+        run: 'ci/ci.sh --platform=x86_64-linux --with-asan --skip-builtins --skip-docs engine'
+      },
+      {
+        name: 'UBSAN',
+        run: 'ci/ci.sh --platform=x86_64-linux --with-ubsan --skip-builtins --skip-docs engine'
+      },
```
Comment (JCash) [inline .github/workflows/engine-nightly.yml:54]: Let's try running it on Linux as well, to possibly catch errors there as well.

File: build_tools/waf_dynamo.py [hunk 3]
```diff
@@ -899,7 +899,7 @@ def authenticode_sign(task):
 
 Task.task_factory('authenticode_sign',
                      func = authenticode_sign,
-                     after = 'link_task stlink_task msvc_manifest')
+                     after = 'link_task stlink_task')
```
Comment (JCash) [inline build_tools/waf_dynamo.py:902]: Fixing waf warnings of missing/unknown dependencies.

File: engine/ddf/src/ddf/wscript [hunk 4]
```diff
@@ -46,6 +46,15 @@ def build(bld):
...
         target = 'ddf')
 
+    bld.add_group() # depend on the pro
+
+    bld.stlib(features = 'cxx ddf skip_asan',
+        includes = '../.. ..',
+        source = 'ddf_extensions.proto ddf_math.proto ddf.cpp ddf_load.cpp ddf_save.cpp ddf_inputbuffer.cpp ddf_util.cpp ddf_message.cpp ddf_loadcontext.cpp ddf_outputstream.cpp',
+        proto_compile_cc = True,
+        protoc_includes = '..',
+        target = 'ddf_noasan')
```
Comment (JCash) [inline engine/ddf/src/ddf/wscript:56]: Create a version of ddf library that doesn't build with asan

File: engine/particle/src/wscript [hunk 5]
```diff
@@ -32,11 +32,11 @@ def build(bld):
     is_host = bld.env['PLATFORM'] in ('x86_64-linux', 'x86_64-win32', 'x86_64-darwin')
 
     if is_host:
-        bld.shlib(features = 'cxx ddf',
+        o = bld.shlib(features = 'cxx ddf skip_asan',
                   includes = ['.', '../proto'],
                   target = 'particle_shared',
                   protoc_includes = '../proto',
-                  uselib = 'DDF DLIB PLATFORM_SOCKET PROFILE_NULL',
+                  use = 'DDF_NOASAN DLIB_NOASAN PLATFORM_SOCKET PROFILE_NULL_NOASAN',
```
Comment (JCash) [inline engine/particle/src/wscript:39]: I think it should be possible to run these shared libraries with asan/ubsan when invoked by Python, but I haven't investigated it. Until then, let's keep the extra non-asan libraries for this purpose.

**Review by Jhonnyg — APPROVED** [review]
LGTM 👍

**Pull request merged**